framework/util/page_guard_manager.h

```diff
@@ -218,16 +228,31 @@ class PageGuardManager
     static PageGuardManager* instance_;
     MemoryInfoMap            memory_info_;
     std::mutex               tracked_memory_lock_;
+    std::mutex               signal_handler_lock_;
     void*                    exception_handler_;
     uint32_t                 exception_handler_count_;
     const size_t             system_page_size_;
     const size_t             system_page_pot_shift_;
     const bool               enable_copy_on_map_;
     const bool               enable_separate_read_;
     const bool               unblock_sigsegv_;
+    bool                     enable_signal_handler_watcher_;
 
     // Only applies to WIN32 builds and Linux/Android builds with PAGE_GUARD_ENABLE_UCONTEXT_WRITE_DETECTION defined.
     const bool enable_read_write_same_page_;
+
+#if !defined(WIN32)
+    pthread_t signal_handler_watcher_thread;
```

Comment:
I think these members should have `_` appended to be consistent with the other members.